Gems/RecastNavigation/Code/Source/Misc/RecastNavigationMeshCommon.cpp

```diff
@@ -323,13 +323,27 @@ namespace RecastNavigation
         return {};
     }
 
+    void RecastNavigationMeshCommon::OnActivate()
+    {
+        m_deactivating = false;
+    }
+
+    void RecastNavigationMeshCommon::OnDeactivate()
+    {
+        m_deactivating = true;
```

Comment:
question: Could this be an enum with more states to make the lifecycle clearer (Activating, Activated, Deactivating, Deactivated) rather than just a bool?

Reply:
The other states don't matter in this context. Really, the meaning is "please cancel async operation".

Reply:
I am renaming it to m_shouldProcessTiles for clarity.

Reply:
Nice one thanks